flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/ResourceManager.java

```diff
@@ -1010,9 +1013,11 @@ private void clearStateInternal() {
      * manager has failed.
      *
      * @param jobId identifying the job whose leader shall be disconnected.
+     * @param resetRequirements whether to reset the resource requirements for the job to 0
      * @param cause The exception which cause the JobManager failed.
      */
-    protected void closeJobManagerConnection(JobID jobId, Exception cause) {
+    protected void closeJobManagerConnection(
+            JobID jobId, boolean resetRequirements, Exception cause) {
```

Comment:
Instead of using a boolean one could also think about an enum. The advantage of an enum would be that one could give an expressive name to the different values.